### ISSUE TITLE
Converting HWCN example convolution kernel to NCHW 

### ISSUE THREAD

**Issue by santhnm2** [dmlc/tvm#1073]
I'm trying to convert the kernel given [here](http://docs.tvmlang.org/tutorials/optimize/opt_conv_cuda.html) to NCHW format. The algorithm specification and blocking strategy seem to be pretty straightforward (just rearranging the axes according to the NCHW format) but I'm having trouble with the cooperative fetching step. Specifically, I'm not sure how to schedule `s[BL]`. The original code splits `rc` and then reorders `s[BL]` as `rco, ry, rx, rci, fi, ni` - is this still what we would want to do for NCHW given the op axis is now `ni, fi, yi, xi`?

**Comment by eqy** [comment]
Loop ordering in general is fairly tricky in this situation because there's many factors at play. Two major things to balance are:
+ memory access stride; generally you want inner loops or those mapped to threads to access adjacent elements so these accesses can be coalesced
+ reuse; if a piece of memory is cached, you want to reuse that piece take advantage of faster cache accesses before moving on

Basically, it depends on how you've set up your cache reads/writes in addition to how you've bound threads to different loop axes. I would try a few different possibilities and experiment as it's pretty difficult to know the precise, best order at a glance :).

**Comment by tqchen** [comment]
good discussion, we have created https://discuss.tvmlang.org for this kind of discussions :)
